### PR TITLE
FW-562. The sixtop transcation should not happen/be aborted if there is not free buffer in schedule

### PR DESCRIPTION
**Pull request by changtengfei**

### CODE REVIEW

File: openstack/02b-MAChigh/schedule.c [hunk 1]
```diff
@@ -293,7 +293,7 @@ owerror_t schedule_addActiveSlot(
    // abort it schedule overflow
    if (slotContainer>&schedule_vars.scheduleBuf[schedule_vars.maxActiveSlots-1]) {
       ENABLE_INTERRUPTS();
-      openserial_printCritical(
+      openserial_printError(
```
Comment (twatteyne) [inline openstack/02b-MAChigh/schedule.c]: great!

File: openstack/02b-MAChigh/schedule.c [hunk 2]
```diff
@@ -557,6 +557,25 @@ uint8_t schedule_getNumOfSlotsByType(cellType_t type){
    return returnVal;
 }
 
+uint8_t schedule_getNumberOfFreeEntries(){
+  uint8_t i;
```
Comment (twatteyne) [inline openstack/02b-MAChigh/schedule.c]: indentation

File: openstack/02b-MAChigh/schedule.h [hunk 3]
```diff
@@ -186,6 +186,7 @@ void              schedule_removeAllCells(
 );
 scheduleEntry_t*  schedule_getCurrentScheduleEntry(void);
 uint8_t           schedule_getNumOfSlotsByType(cellType_t type);
+uint8_t           schedule_getNumberOfFreeEntries();
```
Comment (twatteyne) [inline openstack/02b-MAChigh/schedule.h]: `FreeEntries` is the right term, they are `OffCells`

File: openstack/02b-MAChigh/sixtop.c [hunk 4]
```diff
@@ -1110,7 +1116,7 @@ void sixtop_notifyReceiveCommand(
                         code = IANA_6TOP_RC_SUCCESS;
                         len += processIE_prepend_sixCelllist(response_pkt,cellList);
                     } else {
-                        code = IANA_6TOP_RC_ERR;
+                        code = IANA_6TOP_RC_RESET;
```
Comment (twatteyne) [inline openstack/02b-MAChigh/sixtop.c]: that's not the right status code to send, please see draft.

File: openstack/02b-MAChigh/sixtop.c [hunk 5]
```diff
@@ -175,6 +175,12 @@ void sixtop_request(uint8_t code, open_addr_t* neighbor, uint8_t numCells){
         return;
     }
    
+   if (schedule_getNumberOfFreeEntries() < numCells){
```
Comment (changtengfei) [inline openstack/02b-MAChigh/sixtop.c]: @twatteyne this is the request side changes.

Comment (twatteyne) [inline openstack/02b-MAChigh/sixtop.c]: What I mean is: what does the mote initiating the 6P transaction do when it receives the error code. Does it start over immediately?

**Comment by changtengfei** [comment]
This fix in this PR is already integrated into develop branch. @twatteyne the changes you request is also solved: https://github.com/openwsn-berkeley/openwsn-fw/blob/develop/openstack/02b-MAChigh/sixtop.c#L1249 to handle the RC_NORES return code. Hence, I will close this PR.

**Pull request closed without merging**